rest/client.go

```diff
@@ -164,6 +164,12 @@ func SetDDIAPI() func(*Client) {
 // non-2XX response.
 func (c Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	resp, err := c.httpClient.Do(req)
+
+	if resp != nil {
```

Comment:
Can we move this check below the `if err != nil` beneath this? I don't think there's a reason to eschew Go's convention for error checking immediately after the func call, since HTTP error status code does not result in `httpClient.Do` returning an `err`, right?

Reply:
If the "doer" doesn't return any errors for the requests that have responses, it makes sense to move it down after the first error checking. I am testing, if everything works well I will make another commit moving it

Reply:
I moved since it seems what you said is correct, so it makes senses to keep it after the first error checking